cobra/io/dict.py

```diff
@@ -121,7 +122,16 @@ def reaction_to_dict(reaction):
     new_reaction = OrderedDict()
     for key in _REQUIRED_REACTION_ATTRIBUTES:
         if key != "metabolites":
-            new_reaction[key] = _fix_type(getattr(reaction, key))
+            if key == "lower_bound" and (
+                    np.isnan(reaction.lower_bound) or
```

Comment:
`not np.isfinite(reaction.lower_bound)` is the test you want; https://numpy.org/doc/stable/reference/generated/numpy.isfinite.html